packages/@aws-cdk/aws-elasticloadbalancingv2/lib/alb/application-listener-rule.ts

```diff
@@ -160,6 +171,214 @@ export interface RedirectResponse {
   readonly statusCode: string;
 }
 
+/**
+ * Interface for a listener rule condition
+ */
+export abstract class ListenerRuleCondition {
+  /**
+   * Create a host-header listener rule condition
+   *
+   * @param values Hosts for host headers
+   */
+  public static hostHeaders(values: string[]): ListenerRuleCondition {
+    return new HostHeaderListenerRuleCondition(values);
+  }
+
+  /**
+   * Create a http-header listener rule condition
+   *
+   * @param name HTTP header name
+   * @param values HTTP header values
+   */
+  public static httpHeader(name: string, values: string[]): ListenerRuleCondition {
+    return new HttpHeaderListenerRuleCondition(name, values);
+  }
+
+  /**
+   * Create a http-request-method listener rule condition
+   *
+   * @param values HTTP request methods
+   */
+  public static httpRequestMethods(values: string[]): ListenerRuleCondition {
+    return new HttpRequestMethodListenerRuleCondition(values);
+  }
+
+  /**
+   * Create a path-pattern listener rule condition
+   *
+   * @param values Path patterns
+   */
+  public static pathPatterns(values: string[]): ListenerRuleCondition {
+    return new PathPatternListenerRuleCondition(values);
+  }
+
+  /**
+   * Create a query-string listener rule condition
+   *
+   * @param values Query string key/value pairs
+   */
+  public static queryStrings(values: QueryStringConditionProps[]): ListenerRuleCondition {
+    return new QueryStringListenerRuleCondition(values);
+  }
+
+  /**
+   * Create a source-ip listener rule condition
+   *
+   * @param values Source ips
+   */
+  public static sourceIps(values: string[]): ListenerRuleCondition {
+    return new SourceIpListenerRuleCondition(values);
+  }
+
+  /**
+   * Listener condition field type
+   */
+  public readonly abstract field: string;
+
+  /**
+   * Render the raw Cfn listener rule condition object.
+   */
+  public abstract renderRawCondition(): any;
+}
+
+/**
+ * Properties for the key/value pair of the query string
+ */
+export interface QueryStringConditionProps {
+  /**
+   * The query string key for the condition
+   *
+   * @default - Any key can be matched.
+   */
+  readonly key?: string;
+
+  /**
+   * The query string value for the condition
+   */
+  readonly value: string;
+}
+
+/**
+ * Host header config of the listener rule condition
+ */
+class HostHeaderListenerRuleCondition extends ListenerRuleCondition {
+  public readonly field = 'host-header';
```

Comment:
I don't believe `field` is ever used outside this class anymore, is it?

Reply:
agree! removed it